ledger/participant-integration-api/src/main/scala/platform/store/dao/EventProjectionProperties.scala

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao
+
+import com.daml.ledger.api.domain
+import com.daml.ledger.api.domain.Filters
+import com.daml.lf.data.Ref.{Identifier, Party}
+import com.daml.platform.store.dao.EventProjectionProperties.RenderResult
+
+/** @param verbose enriching in verbose mode
+  * @param populateContractArgument populate contract_argument, and contract_key. If templateId set is empty: populate.
+  * @param populateInterfaceView populate interface_views. The Map of templates to interfaces,
+  *                              and the set of implementor templates cannot be empty.
+  */
+final case class EventProjectionProperties private[dao] (
+    verbose: Boolean,
+    // Map(eventWitnessParty, Set(templateId))
+    populateContractArgument: Map[String, Set[Identifier]] = Map.empty,
+    // Map(eventWitnessParty, Map(templateId -> Set(interfaceId)))
+    populateInterfaceView: Map[String, Map[Identifier, Set[Identifier]]] = Map.empty,
+) {
+  def render(witnesses: Seq[String], templateId: Identifier): RenderResult = {
+    val renderContractArguments = witnesses.view
+      .flatMap(populateContractArgument.get)
+      .exists(templates => templates.isEmpty || templates(templateId))
+
+    val renderInterfaces: Set[Identifier] =
+      (for {
+        witness <- witnesses.iterator
+        templateToInterfaceMap <- populateInterfaceView.get(witness).iterator
+        interfaces <- templateToInterfaceMap.getOrElse(templateId, Set.empty[Identifier])
+      } yield interfaces).toSet
+
+    RenderResult(renderContractArguments, renderInterfaces)
```

Comment:
I think it does more justice to its true intent
```suggestion
    RenderResult(renderContractArguments, interfacesToRender)
```